### ISSUE TITLE
Connect to the internet using curl library

### ISSUE THREAD

**Issue by natruffles** [EECE3093C/under-the-c-desktop#22]
This is necessary to do before interfacing with the OpenWeatherAPI

**Comment by natruffles** [comment]
Hi Jason, I heard you were having issues using the curlpp library to get a text file from a website. I would recommend checking out [example 22](https://github.com/datacratic/curlpp/blob/master/examples/example22.cpp) which has the most basic examples that would apply to our situation.

Use the following URL for your test, and I know it will work:
[https://api.openweathermap.org/data/2.5/weather?q=Cincinnati,US&APPID=095156f9188a2873b0a55d7981778817](https://api.openweathermap.org/data/2.5/weather?q=Cincinnati,US&APPID=095156f9188a2873b0a55d7981778817)

**Comment by natruffles** [comment]
Also, use [this link here](https://stackoverflow.com/questions/662976/how-do-i-convert-from-stringstream-to-string-in-c) for how to convert stringstream output (which is what is generated by curlpp) into a string.

You can then use the [JSON(https://github.com/nlohmann/json) library that Zach is working with to parse that string into a JSON variable to easily pull information (such as zipcode in your case).

**Comment by marti6j8** [comment]
The example is incorrect because cURLpp is a C++ wrapper that can't work independently from libcURL. I found the link to download libcURL at https://curl.se/download.html

**Comment by marti6j8** [comment]
These are the instructions that I received from ChatGPT to download the library
`To use the curl/curl.h library in C++, you need to download and install the cURL library. Here are the steps to download and install the cURL library on Windows:

Go to the cURL website at https://curl.se/download.html.

Scroll down to the "Win64 - Generic" section and click on the latest version of the cURL library to download it.

Extract the contents of the downloaded ZIP file to a directory on your computer. For example, you can extract it to C:\curl-7.80.0-win64-mingw.

Open your C++ project in an IDE like Visual Studio Code or Visual Studio.

Add the path to the include directory of the cURL library to your C++ project's include path. This will allow you to include the curl/curl.h header file in your C++ code. The exact method for doing this will depend on your IDE.

Link your C++ project with the libcurl library file. This will allow you to use the functions defined in the cURL library in your C++ code. The method for doing this will depend on your IDE.

With these steps, you should be able to use the curl/curl.h library in your C++ code.`

**Comment by natruffles** [comment]
This issue should be resolved first, before issue #7.

**Comment by natruffles** [comment]
Hi Zach and Jason,
I think the issue now is that we are not using CMake to link the curl libary with our application, it is currently just a folder sitting there until we use CMake to link them together. I got some great output from ChatGPT. Pictured below are the image responses I got, along with the copyable code at the bottom:
"""""""""""""""""""""""
Here's an example CMakeLists.txt file that you can use to build the libcurl library in C++:
```scss
cmake_minimum_required(VERSION 3.5)
project(libcurl_example)

# Set the C++ standard to use
set(CMAKE_CXX_STANDARD 11)

# Find the libcurl library
find_package(CURL REQUIRED)

# Add the include directory for the libcurl library
include_directories(${CURL_INCLUDE_DIRS})

# Create a library target for libcurl
add_library(curl SHARED IMPORTED)
set_target_properties(curl PROPERTIES IMPORTED_LOCATION ${CURL_LIBRARIES})

# Create an executable target for your program
add_executable(my_program main.cpp)

# Link the executable with the libcurl library
target_link_libraries(my_program curl)
```
In this example, we first set the minimum version of CMake that is required to 3.5. We then define the project name and set the C++ standard to use.

Next, we use the find_package command to locate the libcurl library on your system. We then add the include directory for the libcurl library to the include path of our project using the include_directories command.

We create a library target for libcurl using the add_library command, specifying the SHARED option to create a shared library. We set the properties of the library target using the set_target_properties command, specifying the location of the libcurl library using the IMPORTED_LOCATION property.

Finally, we create an executable target for our program using the add_executable command and link it with the libcurl library using the target_link_libraries command.
""""""""""""""""""""""""""""""""

Hope this info helps. 
If you want to get it from ChatGPT yourself, type in the prompt:
"create a cmakelists.txt file that builds the libcurl library in c++"
